### PR TITLE
Remove ORIGIN macros, make ROM depependency

### PR DESCRIPTION
**Pull request by drichmond**
* Added hb_mc_manycore_eva_init function
* Added prints to dpi_top
* Compute NoC Coordinate widths to match RTL

Derived (and slightly modified) from #752

### CODE REVIEW

**Comment by drichmond** [comment]
Passes regression

File: machine.mk [hunk 1]
```diff
@@ -38,7 +38,7 @@ endif
 
 # To switch machines, simply switch the path of BSG_MACHINE_PATH to
 # another directory with a Makefile.machine.include file.
-BSG_MACHINE_PATH ?= $(BSG_F1_DIR)/machines/pod_X1Y1_ruche_X16Y8_hbm_one_pseudo_channel
```
Comment (dpetrisko) [inline machine.mk:41]: revert default?

Comment (drichmond) [reply machine.mk:41]: Yep, my bad

Comment (drichmond) [reply machine.mk]: Fixed

**Review by dpetrisko — APPROVED** [review]
LGTM

**Pull request merged**